tdrs-backend/tdpservice/search_indexes/test/test_model_mapping.py

```diff
@@ -0,0 +1,389 @@
+        gender=1,
+        receive_nonssa_benefits=1,
+        receive_ssi=1,
+        relationship_hoh=1,
+        parent_minor_child=1,
+        education_level=1,
+        citizenship_status=1,
+        unearned_ssi=1,
+        other_unearned_income=1,
+    )
+
+    assert submission.id is not None
+
+    search = documents.T3DataSubmissionDocument.search().query(
+        'match',
+        record=record_num
+    )
+    response = search.execute()
+
+    assert response.hits.total.value == 1
+
+
+@pytest.mark.django_db
+def test_can_create_and_index_t4_submission(data_analyst, regional_user):
+    """T4 submissions can be created and mapped."""
+    record_num = fake.uuid4()
+
+    submission = T4.objects.create(
+        record=record_num,
+        rpt_month_year=1,
+        case_number='1',
+        disposition=1,
+        fips_code='1',
+
+        county_fips_code='1',
+        stratum=1,
+        zip_code='1',
+        closure_reason=1,
+        rec_sub_housing=1,
+        rec_med_assist=1,
+        rec_food_stamps=1,
+        rec_sub_cc=1,
+        blank='1'
+    )
+
+    assert submission.id is not None
+
+    search = documents.T4DataSubmissionDocument.search().query(
+        'match',
+        record=record_num
+    )
+    response = search.execute()
+
+    assert response.hits.total.value == 1
+
+
+@pytest.mark.django_db
+def test_can_create_and_index_t5_submission(data_analyst, regional_user):
+    """T5 submissions can be created and mapped."""
+    record_num = fake.uuid4()
+
+    submission = T5.objects.create(
+        record=record_num,
+        rpt_month_year=1,
+        case_number='1',
+        disposition=1,
+        fips_code='1',
+
+        family_affiliation=1,
+        date_of_birth='1',
+        ssn='1',
+        race_hispanic=1,
+        race_amer_indian=1,
+        race_asian=1,
+        race_black=1,
+        race_hawaiian=1,
+        race_white=1,
+        gender=1,
+        rec_oasdi_insurance=1,
+        rec_federal_disability=1,
+        rec_aid_totally_disabled=1,
+        rec_aid_aged_blind=1,
+        rec_ssi=1,
+        marital_status=1,
+        relationship_hoh=1,
+        parent_minor_child=1,
+        needs_of_pregnant_woman=1,
+        education_level=1,
+        citizenship_status=1,
+        countable_month_fed_time=1,
+        countable_months_state_tribe=1,
+        employment_status=1,
+        amount_earned_income=1,
+        amount_unearned_income=1
+    )
+
+    assert submission.id is not None
+
+    search = documents.T5DataSubmissionDocument.search().query(
+        'match',
+        record=record_num
+    )
+    response = search.execute()
+
+    assert response.hits.total.value == 1
+
+
+@pytest.mark.django_db
+def test_can_create_and_index_t6_submission(user, deactivated_user):
+    """T6 submissions can be created and mapped."""
+    record_num = fake.uuid4()
+
+    submission = T6.objects.create(
+        record=record_num,
+        rpt_month_year=1,
+        case_number='1',
```

Comment:
These tests are fine and no issue merging as-is. Food for thought though, if we could just use a list of all fields/keys for each document (e.g. `T1.keys()`) could these create calls be simplified? I think such a mechanism might also be reusable for the other areas of the non-test code as well. this could be spun off into a separate ticket, just wanting your thoughts on is it worth doing this?

Some garbage psuedo-code.
```
[myfancydatastructure.append(_.__name__=1) for _ in documents.T6.keys()]
T6.objects.create(**my_fancy_data_structure)
```

Reply:
i like this suggestion a lot :) will get to it